packages/react-aria-components/src/DatePicker.tsx

```diff
@@ -73,7 +73,7 @@ export const DatePickerStateContext = createContext<DatePickerState | null>(null
 export const DateRangePickerStateContext = createContext<DateRangePickerState | null>(null);
 
 // Contexts to clear inside the popover.
-const CLEAR_CONTEXTS = [GroupContext, ButtonContext, LabelContext, TextContext];
+const CLEAR_CONTEXTS = [GroupContext, ButtonContext, LabelContext, TextContext, OverlayTriggerStateContext];
```

Comment:
DateRangePicker too?

What was this needed for? Seems like it might be nice to be able to programmatically close from within the popover